extension/observer/dockerobserver/extension.go

```diff
@@ -36,6 +43,140 @@ func (d *dockerObserver) Shutdown(ctx context.Context) error {
 	return nil
 }
 
+// endpointsForContainer generates a list of observer.Endpoint given a Docker ContainerJSON.
+// This function will only generate endpoints if a container is in the Running state and not Paused.
+func (d *dockerObserver) endpointsForContainer(c *dtypes.ContainerJSON) []observer.Endpoint {
+	cEndpoints := make([]observer.Endpoint, 0)
+
+	if c.State.Running && !c.State.Paused {
```

Comment:
I think that would be fine but I do think it makes it slightly less readable since we're checking for 1 of 4 possible states, so inverted I think it's something like
```    if !c.State.Running || c.State.Running && c.State.Paused {
        return cEndpoints
    }```